openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/retentionPolicyApp/RetentionPolicyApp.java

```diff
@@ -0,0 +1,121 @@
+package org.openmetadata.service.apps.bundles.retentionPolicyApp;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.function.Supplier;
+import lombok.extern.slf4j.Slf4j;
+import org.openmetadata.common.utils.CommonUtil;
+import org.openmetadata.schema.entity.app.App;
+import org.openmetadata.schema.entity.applications.configuration.internal.RetentionPolicyConfiguration;
+import org.openmetadata.service.Entity;
+import org.openmetadata.service.apps.AbstractNativeApplication;
+import org.openmetadata.service.jdbi3.CollectionDAO;
+import org.openmetadata.service.search.SearchRepository;
+import org.openmetadata.service.util.JsonUtils;
+import org.quartz.JobExecutionContext;
+
+@Slf4j
+public class RetentionPolicyApp extends AbstractNativeApplication {
+  private static RetentionPolicyConfiguration retentionPolicyConfiguration;
+
+  public RetentionPolicyApp(CollectionDAO collectionDAO, SearchRepository searchRepository) {
+    super(collectionDAO, searchRepository);
+  }
+
+  @Override
+  public void init(App app) {
+    super.init(app);
+    retentionPolicyConfiguration =
+        JsonUtils.convertValue(app.getAppConfiguration(), RetentionPolicyConfiguration.class);
+  }
+
+  @Override
+  public void startApp(JobExecutionContext jobExecutionContext) {
+    executeCleanup(retentionPolicyConfiguration);
+  }
+
+  public void executeCleanup(RetentionPolicyConfiguration retentionPolicyConfiguration) {
+    if (CommonUtil.nullOrEmpty(retentionPolicyConfiguration)) {
+      return;
+    }
+
+    RetentionPolicyConfiguration.Entity entity = retentionPolicyConfiguration.getEntity();
+    RetentionPolicyConfiguration.RetentionPeriod retentionPeriod =
+        retentionPolicyConfiguration.getRetentionPeriod();
+
+    switch (entity) {
+      case EVENT_SUBSCRIPTION -> eventSubscriptionCleanupHandler(retentionPeriod);
+      default -> throw new IllegalArgumentException("Unknown entity type: " + entity);
+    }
+  }
+
+  private long getRetentionPeriodInMilliseconds(
```

Comment:
This type of hard-coding is not going to work for different entities
1. changeEvents, should have days, weeks
2. Admin can set to retain last 1 day or 2 ,3 days or in weeks
3. Versions, can have retain latest 10 versions, that means anything above latest 10 versions should be deleted
4. Activity threads can be set as retain last month

Reply:
Provided the following retention configurations to be selected:
1. Change Event (Days)
2. Versions (Versions)
3. Activity Threads (Days)